Assets/Scripts/R&D Tree/ResearchNodeSetting.cs

```diff
@@ -0,0 +1,29 @@
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class ResearchNodeSetting : MonoBehaviour
+{
+    [SerializeField] private string _title;
+    [SerializeField] private string _description;
+    [SerializeField] private int _cost;
+    [SerializeField] private int _researchTime;
+
+    [SerializeField] private List<ResearchNode> _connectedResearchNodes;
+    [SerializeField] private List<GameObject> _objectsToUnlock;
```

Comment:
Unlocking is currently bound to GameObjects, but what if I want a variable to change instead of a GameObject... enabled...?

Reply:
And since the `UnlockObjects()` function is empty, I have a feeling this functionality is not yet tested. Try adding something like an event that's called when the node is researched, which an external module can assign `Actions` to. The other module can decide what to do when the event is called.